File: 0675-38-2460-apply-operations-to-an-array/0675-38-2460-apply-operations-to-an-array.py
class Solution:
    def applyOperations(self, nums: List[int]) -> List[int]:
        j = 0
        for i in range(0,len(nums)):
            if i+1 < len(nums) and nums[i] == nums[i+1] and nums[i] > 0:
                x = nums[i]
                nums[i] = nums[i+1] = 0
                nums[j] = x*2
                j+=1
            elif nums[i] > 0:
                nums[i],nums[j] = nums[j],nums[i]
                j+=1
        return nums

File: 0675-38-2460-apply-operations-to-an-array/0675-38-2460-apply-operations-to-an-array.js
function applyOperations(nums) {
    let j = 0;
    const n = nums.length;
    for (let i = 0; i < n; i++) {
        if (i + 1 < n && nums[i] === nums[i + 1] && nums[i] !== 0) {
            const x = nums[i];
            nums[i] = 0;
            nums[i + 1] = 0;
            nums[j++] = x * 2;
        } else if (nums[i] !== 0) {
            [nums[i], nums[j]] = [nums[j], nums[i]];
            j++;
        }
    }
    return nums;
}